### PR TITLE
Add docs for prepare-offline-pack

### PR DESCRIPTION
**Pull request by dedemorton**
Please respond to the questions for Reviewers that I've flagged as comments.

### CODE REVIEW

**Comment by dedemorton** [comment]
This PR addresses the doc changes for offline plugin management tracked in #6501

File: docs/static/offline-plugins.asciidoc [hunk 1]
```diff
@@ -1,17 +1,98 @@
...
+The Logstash <<working-with-plugins,plugin manager>> provides support for preparing offline plugin packs that you can
+use to install Logstash plugins on systems that don't have Internet access. 
 
-The procedures in this section require a staging machine running Logstash that has access to a public or private Rubygems
-server. This staging machine downloads and packages the files used for offline installation.
+This procedure requires a staging machine running Logstash that has access to a public or
+<<private-rubygem,private Rubygems>> server. The staging machine downloads and packages all the files and dependencies
+required for offline installation.
+
+//REVIEWERS: I'm assuming that a private gem repository is still an option, but I haven't tested it. Is it still an option?
```
Comment (ph) [inline docs/static/offline-plugins.asciidoc]: Yes it still an option, we didn't change anything for that, but the prepare-offline-pack should be easier to use than configuring a private gem server.

File: docs/static/offline-plugins.asciidoc [hunk 2]
```diff
@@ -1,17 +1,98 @@
...
+. Run the `bin/logstash-plugin install` subcommand to install the packaged plugins:
++
+["source","sh",subs="attributes"]
+-------------------------------------------------------------------------------
+bin/logstash-plugin install file:///path/to/logstash-offline-plugins-{logstash_version}.zip
+-------------------------------------------------------------------------------
++
+Where +path/to/logstash-offline-plugins-{logstash_version}.zip+ is the path to the offline plugin pack.
+
+//REVIEWERS: Are there any other options from the `install` command that are also supported for an offline plugins install?
```
Comment (ph) [inline docs/static/offline-plugins.asciidoc]: No, not other options are supported or needed at this time.

File: docs/static/offline-plugins.asciidoc [hunk 3]
```diff
@@ -1,17 +1,98 @@
...
+
+//REVIEWERS: In the --help for the latest snapshot build, it doesn't mention that the  --local is deprecated. Looks like the text for the help needs to be updated. Should I open an issue for this, or is this known?
```
Comment (ph) [inline docs/static/offline-plugins.asciidoc]: You are right `--local`  with the install context should also be deprecated, I've took the liberty to create an issue to display a warning see https://github.com/elastic/logstash/issues/6539

**Comment by ph** [comment]
@dedemorton I've answered all the reviewers issues.

**Comment by suyograo** [comment]
LGTM

**Comment by ph** [comment]
LGTM

**Pull request closed without merging**